docs/import/manage_disaster.js

```diff
@@ -635,34 +637,67 @@ function writeNewDisaster(disasterId, states) {
     setStatus('Error: disaster with that name and year already exists.');
     return Promise.resolve(false);
   }
+  clearStatus();
   const currentData = createDisasterData(states);
   disasterData.set(disasterId, currentData);
 
-  const disasterPicker = $('#disaster-dropdown');
-  const disasterOptions = disasterPicker.children();
-  let added = false;
-  // We expect this recently created disaster to go near the top of the list, so
-  // do a linear scan down.
-  // Note: let's hope this tool isn't being used in the year 10000.
-  // Comment needed to quiet eslint.
-  disasterOptions.each(/* @this HTMLElement */ function() {
-    if ($(this).val() < disasterId) {
-      $(createOptionFrom(disasterId)).insertBefore($(this));
-      added = true;
-      return false;
-    }
-  });
-  if (!added) disasterPicker.append(createOptionFrom(disasterId));
-  toggleState(true);
+  const folderCreationPromises = [];
+  folderCreationPromises.push(
+      getCreateFolderPromise(eeLegacyPathPrefix + disasterId));
+  for (const state of states) {
+    folderCreationPromises.push(
+        getCreateFolderPromise(legacyStateDir + '/' + state));
+  }
+
+  Promise.all(folderCreationPromises).then(() => {
+    const disasterPicker = $('#disaster-dropdown');
+    const disasterOptions = disasterPicker.children();
+    let added = false;
+    // We expect this recently created disaster to go near the top of the list,
+    // so do a linear scan down. Note: let's hope this tool isn't being used in
+    // the year 10000. Comment needed to quiet eslint.
+    disasterOptions.each(/* @this HTMLElement */ function() {
+      if ($(this).val() < disasterId) {
+        $(createOptionFrom(disasterId)).insertBefore($(this));
+        added = true;
+        return false;
+      }
+    });
+    if (!added) disasterPicker.append(createOptionFrom(disasterId));
+    toggleState(true);
 
-  disasterPicker.val(disasterId).trigger('change');
+    disasterPicker.val(disasterId).trigger('change');
+  });
 
   return disasterCollectionReference()
```

Comment:
maybe should return the result of all of the promises? So above do "const eePromiseResult = Promise.all(folderCreationPromises).then(()..." and here do "return Promise.all(eePromiseResult, disasterCollectionReference().doc(disasterId).set(currentData)).then(() => true)?